analytical_engine/apps/assortativity/degree/degree_assortativity_coefficient.h

```diff
@@ -0,0 +1,218 @@
+        std::unordered_map<degree_t, std::unordered_map<degree_t, int>> msg;
+        while (messages.GetMessage(msg)) {
+          for (auto& pair1 : msg) {
+            for (auto& pair2 : pair1.second) {
+              // merge
+              if (ctx.degree_mixing_map.count(pair1.first) == 0 ||
+                  ctx.degree_mixing_map[pair1.first].count(pair2.first) == 0) {
+                ctx.degree_mixing_map[pair1.first][pair2.first] = pair2.second;
+              } else {
+                ctx.degree_mixing_map[pair1.first][pair2.first] += pair2.second;
+              }
+            }
+          }
+        }
+        std::vector<std::vector<double>> degree_mixing_matrix;
+        std::unordered_map<int, degree_t> map;
+        // get degree mixing matrix
+        GetDegreeMixingMatrix(ctx, degree_mixing_matrix, map);
+        ctx.degree_assortativity = ProcessMatrix(degree_mixing_matrix, map);
+
+        std::vector<size_t> shape{1};
+        ctx.set_shape(shape);
+        ctx.assign(ctx.degree_assortativity);
+        VLOG(0) << "degree assortatity: " << ctx.degree_assortativity
+                << std::endl;
+      }
+    }
+  }
+
+ private:
+  void ProcessVertex(const vertex_t& v, const fragment_t& frag, context_t& ctx,
+                     message_manager_t& messages) {
+    degree_t source_degree, target_degree;
+    source_degree = GetDegreeByType(frag, v, ctx.source_degree_type_, ctx);
+    // get all neighbors of vertex v
+    auto oes = frag.GetOutgoingAdjList(v);
+    for (auto& e : oes) {
+      vertex_t neighbor = e.get_neighbor();
+      if (frag.IsOuterVertex(neighbor)) {
+        messages.SyncStateOnOuterVertex(frag, neighbor, source_degree);
+      } else {
+        target_degree =
+            GetDegreeByType(frag, neighbor, ctx.target_degree_type_, ctx);
+        DegreeMixingCount(source_degree, target_degree, ctx);
+      }
+    }
+  }
+
+  degree_t GetDegreeByType(const fragment_t& frag, const vertex_t& vertex,
+                           DegreeType type, context_t& ctx) {
+    bool directed = ctx.directed;
+    bool weighted = ctx.weighted;
+    degree_t res;
+    if (weighted) {  // process weighted graph
+      if (!directed || type == DegreeType::OUT) {
+        auto oes = frag.GetOutgoingAdjList(vertex);
+        res = ComputeWeightedDegree(oes);
+      } else {
+        auto oes = frag.GetIncomingAdjList(vertex);
+        res = ComputeWeightedDegree(oes);
+      }
+    } else {  // process unweighted graph
+      // For GraphScope, the inDegree of the undirected graph may be 0
+      if (!directed) {
+        res = static_cast<degree_t>(frag.GetLocalOutDegree(vertex));
+      } else if (type == DegreeType::IN) {
+        res = static_cast<degree_t>(frag.GetLocalInDegree(vertex));
+      } else {
+        res = static_cast<degree_t>(frag.GetLocalOutDegree(vertex));
+      }
+    }
+    return res;
+  }
+
+  template <typename T>
+  degree_t ComputeWeightedDegree(T adjList) {
```

Comment:
add some comment about the function. If the function is simple, we should use inline